src/algoesup/test.py

```diff
@@ -12,6 +30,13 @@ def test(function: Callable, test_table: list) -> None:
             tuple with: a string (the test case name); one or more values (the inputs to the function);
             the expected output value.
     """
+    if errors := validate_test_table(function, test_table):
+        print("ERROR: invalid test table:")
+        for error in errors:
+            print(error)
+        print(f"{function.__name__} has NOT been tested.")
```

Comment:
Your approach is sound: first validate the test table and only run it if it's OK. I had in mind another approach: If the test table is a list or tuple, go through each test any and report it as failed if it is not a tuple/list, has not enough values, doesn't start with a string. In this way the 'syntactically invalid' test cases would be part of the reported failed tests and make the code simpler, but I'm unsure whether it's a better approach than yours.

Reply:
I was worried that the "syntactically invalid" tests might get mistaken for  the "real" tests indicating false negatives. Clear messaging might negate some of this, but I would still worry about that. This is one of the reasons why I tacked on that "... has NOT been tested at the end"